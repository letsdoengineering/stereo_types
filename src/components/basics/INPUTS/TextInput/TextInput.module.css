.text-input {
    border: none;
    color: var(--slate-gray);
    font-size: 1rem;
    width: 100%;
}
