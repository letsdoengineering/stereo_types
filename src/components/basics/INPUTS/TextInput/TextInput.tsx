import React, { forwardRef } from 'react'
import getClassNames from '../../../../utils/get-class-names'
import * as textInputStyles from './TextInput.module.css'

export type TextInputProps = React.ComponentPropsWithRef<'input'> & {
    id?: string | undefined
}

const TextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
    const { id, className, ...rest } = props
    const textInputClassNames = getClassNames({
        defaultClasses: [`${textInputStyles.textInput}`],
        className: className,
    })

    return <input id={id} className={textInputClassNames} type='text' ref={ref} {...rest} />
})
TextInput.displayName = 'TextInput'

export default TextInput
