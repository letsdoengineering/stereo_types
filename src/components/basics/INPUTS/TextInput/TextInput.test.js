import React from 'react'
import TextInput from './TextInput'
import renderer from 'react-test-renderer'

describe('TextInput', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<TextInput />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
