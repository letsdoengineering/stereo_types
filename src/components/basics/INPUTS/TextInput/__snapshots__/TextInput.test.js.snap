// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<input
  className="textInput"
  type="text"
/>
`;
